manifests/config.pp

```diff
@@ -39,4 +49,56 @@
     group   => $otelcol::config_file_group,
     mode    => $otelcol::config_file_mode,
   }
+
+  $otelcol::receivers.each|String $rname, Hash $rvalue| {
+    if $rvalue['config'] =~ Hash {
+      ensure_resource('Otelcol::Receiver', $rname, $rvalue)
+    } else {
+      ensure_resource('Otelcol::Receiver', $rname, { 'config' => $rvalue })
+    }
+  }
+
+  if($otelcol::processors) {
```

Comment:
I'd apply the same as for `$receivers to this and others below it.

Note that `processors` can also be a `String` instead of a `Hash`.

Reply:
Changed that in the same manner as receivers